src/Utils/AccessTrait.php

```diff
@@ -8,17 +8,47 @@
 trait AccessTrait
 {
 	/**
+	 * Transforms objects to arrays
+	 *
+	 * @param $val
+	 * @return mixed
+	 */
+	protected function objectTransform($val)
+	{
+		if (!is_object($val))
+		{
+			return $val;
+		}
+		elseif (is_callable([$val, 'asArray']))
+		{
+			return $val->asArray(true);
+		}
+		else
+		{
+			// Fallback for stdClass objects
+			return json_decode(json_encode($val), true);
```

Comment:
I think array casting with `return (array) $val;` would do the same in less time.